### DISCUSSION TITLE
Does DeepSpeed support tensor parallel for training?

### DISCUSSION THREAD

**Discussion by yiliu30** [deepspeedai/DeepSpeed#4492] [Q&A] [answered]
I understand that Deepspeed supports TP for inference, but it does not support training. 
Can someone share any documentation or demos if I'm incorrect?

**Comment by tjruwase** [comment, marked as answer]
You are correct, DeepSpeed relies on client-side tensor parallelism, such as Megatron, for training. The following doc provides some details on combining DeepSpeed and Megatron for this: https://huggingface.co/blog/bloom-megatron-deepspeed.

**Reply by yiliu30** [reply]
Thanks for your guides, it's a great resource to gain a broad understanding of these technologies.

**Comment by HeatherHong** [comment]
I have the same problem but i didnt find the ways that use only dp and tp automatically

**Reply by inkcherry** [reply]
Hi， @HeatherHong , TP+DP support for HFTrainer is supported, the doc will be added soon

**Reply by JaumePrats** [reply]
Any news on this matter?

**Reply by delock** [reply]
AutoTP training is now supported in DeepSpeed.
https://github.com/deepspeedai/DeepSpeed/blob/master/blogs/huggingface-tp/README.md
